app/utils/ErrorHandler.scala

```diff
@@ -59,7 +59,7 @@ class ErrorHandler @Inject() (
         Future.successful(NotFound(NotFoundError.asJson))
       case _ =>
         val errorCode = statusCode match {
-          case UNAUTHORIZED           => ClientNotAuthenticatedError
+          case UNAUTHORIZED           => ClientOrAgentNotAuthorisedError.withStatus401
           case UNSUPPORTED_MEDIA_TYPE => InvalidBodyTypeError
```

Comment:
in corresponding shared code there is also 
```
 case METHOD_NOT_ALLOWED     => InvalidHttpMethodError
```
https://github.com/hmrc/self-assessment-bsas-api/blob/9f0ffa9931943f171200a4cab4c346380b7baac8/app/shared/utils/ErrorHandler.scala#L70

Reply:
We shouldn’t be adding it where it doesn’t exist.. it can be a breaking change..